### PR TITLE
change the api/core tests to use `core_testing` package

### PR DESCRIPTION
**Pull request by djosephsen**

### CODE REVIEW

**Comment by djosephsen** [comment]
while I was at it, expanded the testcase descriptions to make it more obvious what's being tested

File: api/core/key_test.go [hunk 1]
```diff
@@ -1,31 +1,33 @@
-package core
+package core_test
```
Comment (thinkerou) [inline api/core/key_test.go]: suggestion: use `core` package name

Comment (jmacd) [reply api/core/key_test.go]: By using `core_test` we ensure that we're only testing public interfaces.

Comment (djosephsen) [reply api/core/key_test.go]: ☝️

**Pull request merged**